### PR TITLE
One rewind data instance per block when close to chain tip

### PR DESCRIPTION
**Pull request by noescape00**
Flush coinview when consensus tip is close to the chain tip to ensure that we have 1 rewind data per block. 
Previously we were creating 1 rewind data for batch of blocks so during a reorg at the tip we were able to call a rewind on consensus tip and it would rewind a lot of blocks (up to 5000 if reorg happened when we just synced).

This is PR 2 from the list of PRs for the big refasctoring

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs [hunk 1]
```diff
@@ -441,6 +439,22 @@ public void Stop()
...
+        /// <remarks>
+        /// For blockchains with max reorg property flush is required when consensus tip is less than max reorg blocks behind the chain tip.
+        /// If there is no max reorg property - flush is required when consensus tip timestamp is less than two weeks behind the adjusted time.
+        /// </remarks>
+        private bool FlushRequired()
+        {
+            if (this.chainState.MaxReorgLength != 0)
+                return this.Chain.Height - this.Tip.Height < this.chainState.MaxReorgLength;
+
+            int twoWeeksSeconds = 2 * 24 * 60 * 60;
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs]: `twoWeeksSeconds` -> `twoWeeksAsSeconds`

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs]: Two weeks might be too long no? even 1 day is enough.

**Comment by dangershony** [comment]
Ah I did not run tests yet (or sync chains).

**Comment by noescape00** [comment]
>Ah I did not run tests yet (or sync chains).

Kk. 

Ps
For me tests are passing (except for those tests that fail for 2 weeks already), stratis main and test synced.

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs [hunk 2]
```diff
@@ -441,6 +439,22 @@ public void Stop()
...
+        /// <remarks>
+        /// For blockchains with max reorg property flush is required when consensus tip is less than max reorg blocks behind the chain tip.
+        /// If there is no max reorg property - flush is required when consensus tip timestamp is less than two weeks behind the adjusted time.
+        /// </remarks>
+        private bool FlushRequired()
+        {
+            if (this.chainState.MaxReorgLength != 0)
+                return this.Chain.Height - this.Tip.Height < this.chainState.MaxReorgLength;
+
+            int twoWeeksAsSeconds = 2 * 24 * 60 * 60;
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs]: This should be a constant on the class with comment. It should be explained why this number was selected.

Comment (noescape00) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs]: Fixed

**Pull request merged**